src/process_utility.c

```diff
@@ -233,11 +234,51 @@ check_alter_table_allowed_on_ht_with_compression(Hypertable *ht, AlterTableStmt
 			case AT_AddColumn:	 /* this is passed down */
 			case AT_ColumnDefault: /* this is passed down */
 			case AT_DropColumn:	/* this is passed down */
+
+			case AT_DropConstraint:
+			case AT_AlterConstraint:
 #if PG14_GE
 			case AT_ReAddStatistics:
 			case AT_SetCompression:
 #endif
 				continue;
+
+			/*
+			 * Conditionally allowed command. Adding a unique constraint
+			 * including non-segmentby column is not supported.
+			 */
+			case AT_AddConstraint:
+			{
+				Constraint *constraint = (Constraint *) cmd->def;
+
+				if (constraint->contype == CONSTR_UNIQUE)
+				{
+					List *segmentby = NIL, *info = NIL;
+					ListCell *lc;
+
+					info = ts_hypertable_compression_get(ht->fd.id);
+					foreach (lc, info)
+					{
+						FormData_hypertable_compression *fd = lfirst(lc);
+						if (fd->segmentby_column_index > 0 ||
+							fd->orderby_column_index > 0)
```

Comment:
```suggestion
						if (fd->segmentby_column_index > 0)
```
Unique constraints including orderby columns are not supported

Reply:
Right, I've confused it with the dimension columns, which have to be included here, otherwise creating a unique constraint still fails later on in indexing.